### PR TITLE
exit the main maxwell thread cleanly

### PR DESCRIPTION
**Pull request by osheroff**
@zendesk/rules

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/MaxwellParser.java [hunk 1]
```diff
@@ -78,18 +80,48 @@ public void start() throws Exception {
...
 
-	public void stop() throws Exception {
+	public void stop() {
+		// note: we use stderr in this function as LOGGER.err() oftentimes
+		// won't flush in time, and we lose the messages.
+
 		this.binlogEventListener.stop();
-		this.replicator.stop(5, TimeUnit.SECONDS);
+		try {
+			this.replicator.stop(5, TimeUnit.SECONDS);
```
Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: this is a bit weird. Considering there is a chance that L129 (https://github.com/zendesk/maxwell/pull/111/files#diff-5adf358ee2ff00689a314b88b7f2786cR129 ) may simply restart it in the main thread. 

I suggest setting the shouldStop flag here, and maybe letting the main thread do this cleanup?

Comment (vanchi-zendesk) [inline src/main/java/com/zendesk/maxwell/MaxwellParser.java]: Or atleast check shouldStop before L129 ?

**Comment by vanchi-zendesk** [comment]
Think the overall approach is fine. :+1: but for the comment above.

**Comment by vanchi-zendesk** [comment]
ready for review again?

**Comment by osheroff** [comment]
nope, working on a bigger refactor that should hopefully clean some of this crap up.  a bit.

**Comment by osheroff** [comment]
Ok, this is ready for review now.  hopefully cleaner.

**Comment by vanchi-zendesk** [comment]
:+1: nice

**Pull request merged**